File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int N = nums.length;
        int start = 0;
        for (int i = 0; i < N; i++) {
            if (nums[i] != 0) {
                nums[start++] = nums[i];
            }
        }

        while (start < N) {
            nums[start++] = 0; 
        }
    }
}
 
/* 
-----------------Runtime High Solution--------------------
class Solution {
    public void moveZeroes(int[] nums) {
        int N = nums.length;
        for (int i = 0; i < N; i++) {
            int curr = nums[i];

            if (curr == 0) {
                int nextIndex = i + 1;
                while (nextIndex < N && nums[nextIndex] == 0) {
                    nextIndex++;
                }

                if (nextIndex < N) {
                    nums[i] = nums[nextIndex];
                    nums[nextIndex] = curr;
                }    
            }
        }
    }
}
*/

File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        start = i = 0

        while i < len(nums):
            if nums[i] != 0:
                nums[start] = nums[i]
                start += 1
            i += 1
        
        for j in range(start, len(nums)):
            nums[j] = 0